packages/true-time/src/index.ts

```diff
@@ -1,39 +1,46 @@
 export class TrueTime {
-  #clientStartTime: number;
+  #clientStartTime?: number;
 
   #serverTime?: number;
 
-  #synced = false;
-
   #url: URL;
 
   constructor(url: string) {
     this.#url = new URL(url);
-    this.#clientStartTime = performance.now();
   }
 
-  now(highResTimeStamp: DOMHighResTimeStamp = performance.now()): number {
+  /**
+   * Returns the current time adjusted to server-time.
+   *
+   * @param clientCurrentTime The current time on the client side. Defaults to Date.now().
+   * @returns The current adjusted time.
+   * @throws {ReferenceError} If the initialization has not been done yet. You need to call and await the `synchronize` method once.
+   */
+  // eslint-disable-next-line no-restricted-syntax
+  now(clientCurrentTime = Date.now()): number {
     if (!this.#serverTime || !this.#clientStartTime) {
       throw new ReferenceError(
         'Initialization has not been done yet. You need to call and await the synchronize method once.',
       );
     }
 
-    return Number.parseInt(
-      (
-        this.#serverTime + Math.abs(this.#clientStartTime - highResTimeStamp)
-      ).toFixed(0),
-      10,
-    );
+    return this.#serverTime + (clientCurrentTime - this.#clientStartTime);
   }
 
   /**
-   * Use this method to synchronize time with the server.
+   * Synchronizes the client's time with the server's time.
+   * If the client's time is already synchronized within an hour, this method does nothing.
    *
-   * @param url - server url
+   * @returns {Promise<void>} A promise that resolves when the synchronization is complete.
    */
-  async synchronize() {
-    if (this.#synced) {
+  async synchronize(): Promise<void> {
+    const anHour = 3_600_000;
```

Comment:
I put 1_000_000 in PR since ChatGPT said this about NTP;

```
As a general guideline, a common default configuration for many systems is to synchronize every 64 to 1,024 seconds (approximately 1 to 17 minutes).
```

1 million ms is 16.67 minutes so figured that was nice. :P

Reply:
Hehe, so that was where you got the number from :-P I figured an hour should be frequent enough..